scripts/zemax-api.js

```diff
@@ -0,0 +1,25 @@
+import { getEnvironmentConfig } from './zemax-config.js';
+
+export default function execute(actionName, urlParams, methodType) {
+  const userId = localStorage.getItem('auth0_id');
+  const accessToken = localStorage.getItem('accessToken');
+  const DYNAMIC_365_DOMAIN = getEnvironmentConfig('dev').profile.dynamic365domain;
+  return new Promise((resolve, reject) => {
+    fetch(`${DYNAMIC_365_DOMAIN}${actionName}?auth0_id=${userId}${urlParams}`, {
+      method: methodType,
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+        Authorization: `bearer ${accessToken}`,
+      },
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        // Process the data if needed
+        resolve(data);
+      })
+      .catch((error) => {
+        console.error('Error:', error);
+        reject(error);
+      });
+  });
+}
```

Comment:
it's better to use async/await syntax, makes to code easier to read. 

quick draft: 
```suggestion
export default async function execute(actionName, urlParams, methodType) {
  try {
    const userId = localStorage.getItem('auth0_id');
    const accessToken = localStorage.getItem('accessToken');
    const DYNAMIC_365_DOMAIN = getEnvironmentConfig('dev').profile.dynamic365domain;

    const response = await fetch(`${DYNAMIC_365_DOMAIN}${actionName}?auth0_id=${userId}${urlParams}`, {
      method: methodType,
      headers: {
        'Content-type': 'application/json; charset=UTF-8',
        Authorization: `bearer ${accessToken}`,
      },
    });

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    // Process the data if needed
    const data = await response.json();
    return data;
  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}
```

Reply:
Updated to your suggested code